### ISSUE TITLE
Analyzer to convert Assert.Collection to Assert.Single

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2123]
While updating the xUnit.net source, I noticed that I did this a lot:

```csharp
Assert.Collection(
    collection,
    item => // assertion(s) for the single item
);
```

This could/should be converted to this:

```csharp
var item = Assert.Single(collection);
// assertion(s) for the single item
```

**Comment by netchkin** [comment]
Can you please include an example for `// assertion(s) for the single item` in `Assert.Collection` example?

**Comment by adison88** [comment]
@bradwilson I see most of these usages in unit tests. I can pick this one up to update the source. Please let me know if there're any other places which needs an update.

**Comment by adison88** [comment]
Hello @bradwilson , created https://github.com/xunit/xunit/pull/2230 for review.

**Comment by vb2ae** [comment]
@brad did pr #2230 fix the issue?  Wondering if this ticket should be closed

**Comment by bradwilson** [comment]
It did. Thanks for pointing it out.

**Comment by bradwilson** [comment]
Oops, I jumped the gun. That PR fixed the usage in xUnit.net, but did not create the analyzer.

**Comment by adison88** [comment]
@bradwilson what other changes are needed to close this ticket?

**Comment by bradwilson** [comment]
> @bradwilson what other changes are needed to close this ticket?

This issue is to create a new analyzer and fixer.

**Comment by etherfield** [comment]
@adison88, do you still want to do this? I can help with the analyzer and fixer if you don't mind.

**Comment by adison88** [comment]
@etherfield go ahead.

**Comment by bradwilson** [comment]
Available in `1.6.0-pre.8`.
